### ISSUE TITLE
aria-hidden being set true for elements in viewport

### ISSUE THREAD

**Issue by geon-halfhelix** [metafizzy/flickity#1228]
This is an amazing library. Kudos to the devs!

I noticed an accessibility issue in the slider. 
The issue is that `aria-hidden` attribute is being set to `true` for elements which are in the slider view.
Please see this reproduction I made: https://codepen.io/geon-halfhelix/pen/YzeEbLj

![image](https://user-images.githubusercontent.com/98684094/170481095-3d74fba9-91ee-4eb1-8791-4233e59fb396.png)

**Comment by neodigm** [comment]
Hi. I too have encountered this issue.  It's failing Chrome Lighthouse A11y audit.

 * Flickity PACKAGED v2.2.2

**Comment by aspatino** [comment]
Hello,

We're experiencing this issue on a site. Any word on a fix?

**Comment by desandro** [comment]
Thanks for reporting this issue.  **Add a 👍  reaction** to this issue if you would like to see this feature added.

The challenge is calculating which cells are within the viewport any time a slide is selected.

**Comment by mdoxwp** [comment]
Am seeing the same issue. Please include/fix in the next build.

**Comment by jimrowetpa** [comment]
I also have this issue, is there a fix incoming or a workaround anyone is aware of? Thanks.

**Comment by adrielklein** [comment]
Same here, would love a fix

**Comment by mangitmaharjan** [comment]
Any solutions?

**Comment by neodigm** [comment]
Hi. I can imagine a solution using [InterSectionObserver](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API). If anyone is interested, I could write a work around patch, or create a PR. Let me know if I can help.

**Comment by pirco** [comment]
my client is being sued because of ADA non-compliance. this issue is a sticking point. I'm interested @neodigm

**Comment by neodigm** [comment]
Hi, I've create a patch that should improve the Lighthouse audit results with one or more carousels on the page:

[Flick aria-hidden patch](https://gist.github.com/neodigm/bf64202df7ff854c910977eb1f2515a0)

I've tried on a few production sites with good results. Let me know if you've any issues.

**Comment by marshall993** [comment]
Reporting this issue as well as it's impacting our Lighthouse score. `86` enabled, `96` disabled. Is there a fix on the way?

**Comment by JGallardo4** [comment]
Is there a fix for this in Flickity 3.0.0?

**Comment by dotjay** [comment]
There is a related issue here that all focusable elements within a parent element that has `aria-hidden="true"` remain keyboard focusable. This means that hidden content must be navigated when using a keyboard, making Flickety carousels very frustrating for keyboard users. It also means that content semantics are not conveyed to assistive technologies when these hidden elements become focussed, making it very confusing to navigate for people using any assistive technology software like screen readers.

I think it's important to think of accessibility issues as critical as this as being a critical bug and not a feature, as this issue is described. When accessibility is considered a feature, it is rarely prioritised, as demonstrated by this issue being nearly 3 years old. While I appreciate there's an open source aspect to this project, which hasn't seen any updates in 2 years, it's also a commercial product that shouldn't be excluding people on the basis of disability.

**Comment by RYJASM** [comment]
Has there been any movement on accessibility improvements? We will have to stop using flickity since the US government has mandated wacag 2.1 accessibility compliance.

**Comment by stefbowerman** [comment]
I fixed this using IntersectionObserver like this:

```typescript
const flckty = new Flickity(...)
const cellObservers = new Map<HTMLElement, IntersectionObserver>()

flckty.cells.forEach((cell) => {
  const observer = new IntersectionObserver((entries) => {
    const entry = entries[0]
    if (entry?.isIntersecting) {
      cell.element.setAttribute('aria-hidden', entry?.isIntersecting ? 'false' : 'true')
    }
  }, { threshold: 0.01 })

  cellObservers.set(cell.element, observer)
  observer.observe(cell.element)
})
```

**Comment by jsfgreen** [comment]
> I fixed this using IntersectionObserver like this:
> 
> const flckty = new Flickity(...)
> const cellObservers = new Map<HTMLElement, IntersectionObserver>()
> 
> flckty.cells.forEach((cell) => {
>   const observer = new IntersectionObserver((entries) => {
>     const entry = entries[0]
>     if (entry?.isIntersecting) {
>       cell.element.setAttribute('aria-hidden', entry?.isIntersecting ? 'false' : 'true')
>     }
>   }, { threshold: 0.01 })
> 
>   cellObservers.set(cell.element, observer)
>   observer.observe(cell.element)
> })

This is solid!  The browser still complains if you have links in the cell which are focusable elements inside an `aria-hidden="true"`.

```javascript
const cellObservers = new Map();
flick.cells.forEach(cell => {
    const observer = new IntersectionObserver(entries => {
        const entry = entries[0];
        cell.element.setAttribute('aria-hidden', entry.isIntersecting ? 'false' : 'true');

        if (!entry.isIntersecting) {
            cell.element.querySelector('a')?.setAttribute('tabindex', '-1');
        } else {
            cell.element.querySelector('a')?.removeAttribute('tabindex');
        }
    }, { threshold: 0.01 });

    cellObservers.set(cell.element, observer);
    observer.observe(cell.element);
});
```

Use `querySelectorAll` if you have multiple links.  In my case I only have one usually for - I believe - all my use cases for flickity.  It's not perfect but it solved the lighthouse audit tool issues.  Querying the DOM a lot is the obvious drawback here.  But it's generic enough to be reusable.
